pkg/app/server/apikeyverifier/verifier.go

```diff
@@ -54,37 +56,46 @@ func (v *Verifier) Verify(ctx context.Context, key string) (*model.APIKey, error
 	item, err := v.apiKeyCache.Get(keyID)
 	if err == nil {
 		apiKey = item.(*model.APIKey)
-		if err := checkAPIKey(apiKey, keyID, key); err != nil {
+		if err := v.checkAPIKey(ctx, apiKey, keyID, key); err != nil {
 			return nil, err
 		}
 		return apiKey, nil
 	}
-
 	// If the cache data was not found,
 	// we have to retrieve from datastore and save it to the cache.
 	apiKey, err = v.apiKeyStore.Get(ctx, keyID)
 	if err != nil {
 		return nil, fmt.Errorf("unable to find API key %s from datastore, %w", keyID, err)
 	}
 
+	// update the time API key was last used
+
 	if err := v.apiKeyCache.Put(keyID, apiKey); err != nil {
 		v.logger.Warn("unable to store API key in memory cache", zap.Error(err))
 	}
-	if err := checkAPIKey(apiKey, keyID, key); err != nil {
+	if err := v.checkAPIKey(ctx, apiKey, keyID, key); err != nil {
 		return nil, err
 	}
 
 	return apiKey, nil
 }
 
-func checkAPIKey(apiKey *model.APIKey, id, key string) error {
+func (v *Verifier) checkAPIKey(ctx context.Context, apiKey *model.APIKey, id, key string) error {
 	if apiKey.Disabled {
 		return fmt.Errorf("the api key %s was already disabled", id)
 	}
 
 	if err := apiKey.CompareKey(key); err != nil {
 		return fmt.Errorf("invalid api key %s: %w", id, err)
 	}
+	now := v.now()
+	if err := v.apiKeyLastUsedCache.Put(id, now); err != nil {
+		return fmt.Errorf("unable to update the time API key %s was last used, %w", id, err)
+	}
 
 	return nil
 }
+
+func (v *Verifier) now() int64 {
+	return time.Now().Unix()
+}
```

Comment:
Should make nowFunc a field in Verifier struct, like this.

https://github.com/pipe-cd/pipecd/blob/master/pkg/app/piped/controller/planner.go#L66